### PR TITLE
Fix up expression construction

### PR DESCRIPTION
**Pull request by tpoterba**
depends on #2548 (not a stacked PR because I don't want them to have to go in together)

### CODE REVIEW

File: python/hail/docs/utils/index.rst [hunk 1]
```diff
@@ -3,6 +3,8 @@ utils
 
 .. currentmodule:: hail.utils
 
+.. autoclass:: Struct
```
Comment (jigold) [inline python/hail/docs/utils/index.rst]: Is this the right place for `Struct`?

File: python/hail/methods/statgen.py [hunk 2]
```diff
@@ -0,0 +1,128 @@
...
+        jarray(Env.jvm().java.lang.String, [cov._ast.to_hql() for cov in covariates]),
+        'va.`{}`'.format(root),
+        block_size
+    )
+
+    return cleanup(MatrixTable(dataset._hc, jm))
+
+
+@handle_py4j
+@typecheck(dataset=anytype, force_local=bool)
```
Comment (jigold) [inline python/hail/methods/statgen.py]: Shouldn't the type check check for a MatrixTable?

File: python/hail/utils/__init__.py [hunk 3]
```diff
@@ -0,0 +1,17 @@
+from .summary import Summary
+from .misc import FunctionDocumentation, wrap_to_list, get_env_or_default
+from .hadoop_utils import hadoop_copy, hadoop_read, hadoop_write
+from .struct import Struct
+from .java import error, warn, info
+
+__all__ = ['Summary',
```
Comment (jigold) [inline python/hail/utils/__init__.py]: Should Summary be in hail.genetics?

File: python/hail/methods/statgen.py (hunk 2 above)
Comment (tpoterba) [reply python/hail/methods/statgen.py]: yes, I recall this having circular import problems. Will try to do that.

File: python/hail/utils/__init__.py (hunk 3 above)
Comment (tpoterba) [reply python/hail/utils/__init__.py]: summary should be kaputt. We can fix it in a future PR

File: python/hail/docs/utils/index.rst (hunk 1 above)
Comment (tpoterba) [reply python/hail/docs/utils/index.rst]: hail.utils? I think so. Where else should it be?

File: python/hail/api2/tests.py [hunk 4]
```diff
@@ -620,8 +617,8 @@ def test_constructors(self):
                                        f.locus("1", 56, reference_genome=rg)))
 
         expected_schema = {'a': TFloat64(), 'b': TFloat64(), 'c': TInt32(), 'd': TInt64(), 'v1': TVariant(rg),
-                           'v2': TVariant(rg), 'v3': TVariant(rg), 'l1': TLocus(), 'l2': TLocus(rg), 'i1': TInterval(rg),
+                           'v2': TVariant(rg), 'v3': TVariant(rg), 'l1': TLocus(), 'l2': TLocus(rg),
+                           'i1': TInterval(rg),
```
[the remark this answers is not in the document]
Comment (tpoterba) [reply python/hail/api2/tests.py]: sure.

**Comment by tpoterba** [comment]
These comments were all on the portion in #2548 I think, I'll make the changes there.

Sorry to present the combined diff like this...

File: python/hail/methods/statgen.py (hunk 2 above)
Comment (tpoterba) [reply python/hail/methods/statgen.py]: yep, could drop in the type. Good catch.

**Pull request merged**